pkg/persistence/redis/redisclient.go

```diff
@@ -0,0 +1,160 @@
+package redis
+
+import (
+"errors"
+"fmt"
+"net/url"
+"strconv"
+"strings"
+"sync"
+"time"
+
+"github.com/FZambia/sentinel"
+"github.com/gomodule/redigo/redis"
+)
+
+var knownPool sync.Map
+var m sync.Mutex
+
+// RedisPoolParam ...
+type RedisPoolParam struct {
+	PoolMaxIdle     int
+	PoolMaxActive   int
+	PoolIdleTimeout time.Duration
+
+	DialConnectionTimeout time.Duration
+	DialReadTimeout       time.Duration
+	DialWriteTimeout      time.Duration
+}
+
+// GetRedisPool get a named redis pool
+// supported schema
+// redis://user:pass@redis_host:port/db
+// redis+sentinel://user:pass@redis_sentinel1:port1,redis_sentinel2:port2/monitor_name/db
+func GetRedisPool(name string, rawurl string, param *RedisPoolParam) (*redis.Pool, error) {
+	if p, ok := knownPool.Load(name); ok {
+		return p.(*redis.Pool), nil
+	}
+	m.Lock()
+	defer m.Unlock()
+	// load again in case multi threads
+	if p, ok := knownPool.Load(name); ok {
+		return p.(*redis.Pool), nil
+	}
+
+	u, err := url.Parse(rawurl)
+	if err != nil {
+		return nil, fmt.Errorf("bad redis url: %s", err)
+	}
+
+	if param == nil {
+		param = &RedisPoolParam{
+			PoolMaxIdle:           0,
```

Comment:
As mentioned above, the right place for the default values would be the `etc` package and the `Store` struct. See the existing tags to configure envs